sdk/src/OpenTelemetry/LogRecordProcessor/Simple.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module OpenTelemetry.LogRecordProcessor.Simple (
+  simpleProcessor,
+) where
+
+import Control.Concurrent.Async (async, cancel)
+import Control.Concurrent.Chan.Unagi
+import Control.Exception
+import Control.Monad (forever)
+import qualified Data.HashMap.Strict as H
+import qualified Data.Vector as V
+import OpenTelemetry.Internal.Common.Types
+import OpenTelemetry.Internal.Logs.Types
+
+
+{- | This is an implementation of LogRecordProcessor which passes finished logs and passes the export-friendly ReadableLogRecord
+representation to the configured LogRecordExporter, as soon as they are finished.
+-}
```

Comment:
This implementation mirrors `SpanProcessor.Simple` but with a lot of refactoring. It also adds propagating `ExportResult`s to `FlushResult`s to `ShutdownResult`s.